src/libxrpl/ledger/View.cpp

```
@@ -1282,6 +1282,8 @@ addEmptyHolding(
         return tefINTERNAL;  // LCOV_EXCL_LINE
     if (view.peek(keylet::mptoken(mptID, accountID)))
         return tecDUPLICATE;
+    if (accountID == mptIssue.getIssuer())
+        return tesSUCCESS;
```

is `tesSuccess` really the correct code to return here? From my intuition, an issuer should not be allowed to add an empty holding, therefore an error code should be returned instead - and the caller of `addEmptyHolding` should be responsible for checking that the accountID is not the issuer.

---

I think this is the reason: https://github.com/XRPLF/rippled/pull/5875/files#r2426305173

---

although it's clean to handle this edge case in the function, i still think it might be something that's worth being explicitly handled by the caller for safety practice, as there could be downstream bugs if future callers aren't aware that this edge case is silently returned as tesSuccess. But again, we do have sufficient unit tests and invariants for so the likelihood of something like this going wrong is pretty low.

---

`temDST_IS_SRC` is the result returned by a `TrustSet` transaction if the account is the issuer. I think the situation is analogous enough that it would be appropriate here, too.